src/main/kotlin/com/cognifide/gradle/aem/common/file/IoTransferLogger.kt

```diff
@@ -0,0 +1,94 @@
+package com.cognifide.gradle.aem.common.file
+
+import com.cognifide.gradle.aem.common.Formats
+import com.cognifide.gradle.aem.common.ProgressLogger
+import java.io.File
+import java.io.FileOutputStream
+import java.io.InputStream
+import java.io.OutputStream
+import org.gradle.api.Project
+
+class IoTransferLogger(val project: Project) : IoTransfer() {
```

Comment:
looks like code is copied `IoTransferLogger` and `IoTransfer`

how about having class

```kotlin

open class IoTransfer {

var onUpload: (File) -> Unit = {}

upload() {
// ...
    processedBytes += read
    onUpload(file)
    read = input.read(buf)
// ...
```


then class extending it

```kotlin
class IoLoggedTransfer : IoTransfer {
   onUpload = { file -> logProgress("Uploading", file.length(), file) }
}
```
```

Reply:
is there a place where you are not using logged transfer? why do you want to have 2 versions?

Reply:
I do not want to use in tests. Since it connects testing with Gradle objects.

Reply:
OK, I will unify this using lambda